py/server/tests/test_function_generated_table.py

```diff
@@ -39,20 +42,26 @@ def test_generated_table_1trigger(self):
         append_only_input_table = input_table(col_defs=col_defs)
 
         def table_generator_function():
+            print("Running table_generator_function() at time: " + str(datetime.now()))
             return append_only_input_table.last_by().update('ResultStr = MyStr')
 
         result_table = function_generated_table(table_generator_function, source_tables=append_only_input_table)
 
         self.assertEqual(result_table.size, 0)
 
+        print("Adding row at time: " + str(datetime.now()))
         append_only_input_table.add(new_table([string_col(name='MyStr', data=['test string'])]))
+        print("add() returned at time: " + str(datetime.now()))
         self.wait_ticking_table_update(result_table, row_count=1, timeout=30)
-        self.assertEqual(result_table.size, 1)
+        print("result_table has 1 row at time: " + str(datetime.now()))
```

Comment:
So the row somehow vanished. Is the table a blink table?

Reply:
nah — it's an input table, which is backed by an AppendOnlyArrayBackedMutableTable